proposals/lip-0031.md

```diff
@@ -191,7 +195,7 @@ proof = {
 In particular:
 
 * `size` is encoded as a `uint64`.
-* Each index in `idxs` is prepended a `1` and is encoded as a `uint64`. The extra `1` is added as a delimiter to set the correct index length. At the moment of decoding, each index is converted from unsigned integer to binary, the first bit (which equals `1`) is dropped and only the following bits are kept. Furthermore, we use the special index `0` to flag any element which is not part of the data array, and as such can be discarded during the verification. Notice that this does not constitute a proof-of-absence for that element.
+* Each index in `idxs` represented as a bit string is prepended a `1` and encoded as a `uint64` (see [above](#representation-of-indices) for the reasoning). Whenever an index needs to be converted from its `unit64` representation to a binary string, the bit string following the most significant bit must be taken. Furthermore, we use the special index `0` to flag any element which is not part of the data array, and as such can be discarded during the verification. Notice that this does not constitute a proof-of-absence for that element.
 * The elements of `siblingHashes` are encoded as an array of `bytes` of length 32.
 
 The hex-encoded proof for the example in Fig. 1 is `proof=0x0805||0x120111||0x1a20||h0||0x1a20||h6||0x1a20||h4`, with a length `L=107` bytes.
```

Comment:
I am not sure that this example is correct, but I might be missing something. In particular I don't understand the reason why the first two values are set like that.

Reply:
I didn't check all values, but I would say that `0x0805` is the encoding of the size, and `0x120111` of `idxs` which is an array of length one.